lib/hiera.rb

```diff
@@ -31,8 +31,13 @@ def logger=(logger)
       warn("Failed to load #{logger} logger: #{e.class}: #{e}")
     end
 
-    def warn(msg); @logger.warn(msg); end
-    def debug(msg); @logger.debug(msg); end
+    def warn(msg = '', &block)
+      block_given? ? @logger.debug(&block) : @logger.debug(msg)
```

Comment:
I think it should raise an error if both a message and a block was specified at the same time since the message will be ignored in that case.